src/ServiceV2/GdprService.php

```diff
@@ -30,16 +31,16 @@ public function isPasswordRenewalDue(): bool
         return $this->getDaysBeforeExpiration() <= self::RENEWAL_DAYS_COUNT;
     }
 
-    public function isPasswordExpired(): bool
+    public function isPasswordExpired(User $user = null): bool
     {
-        return $this->getDaysBeforeExpiration() <= self::EXPIRATION_DAYS_COUNT;
+        return $this->getDaysBeforeExpiration($user) <= self::EXPIRATION_DAYS_COUNT;
     }
 
-    private function getDaysBeforeExpiration(): int
+    private function getDaysBeforeExpiration(User $user = null): int
     {
         return max((int) (new \DateTimeImmutable())
             ->sub(new \DateInterval('P1Y'))
-            ->diff($this->getUser()->getPasswordUpdatedAt() ?? new \DateTimeImmutable())
+            ->diff($user ?: $this->getUser()->getPasswordUpdatedAt() ?? new \DateTimeImmutable())
```

Comment:
I'm not sure this is a correct condition
`$user?->getPasswordUpdatedAt ?? $this->getUser()->getPasswordUpdatedAt() ?? new \DateTimeImmutable()`